### PR TITLE
(FM-8496) Add support for Puppet debug flags during /execute_catalog

### PR DESCRIPTION
**Pull request by david22swan**

### CODE REVIEW

File: spec/unit/ace/transport_app_spec.rb [hunk 1]
```diff
@@ -308,6 +318,59 @@ def app
       end
     end
 
+    context 'when given flag debug' do
+      it 'returns 200 with success' do
```
Comment (da-ar) [inline spec/unit/ace/transport_app_spec.rb]: you probably want to add:

```
expect(last_response).to be_ok
expect(last_response.status).to eq(200)
expect(last_response.status).to eq(200)
```

back in

File: spec/unit/ace/transport_app_spec.rb [hunk 2]
```diff
@@ -308,6 +318,79 @@ def app
...
+      end
+    end
+
+    context 'when given flag noop' do
+      it 'returns 200 with success' do
+        execute_catalog_body[:compiler][:noop] = true
+
+        expect(psettings).to receive(:[]=).with(:noop, true)
+        expect(psettings).to receive(:[]=).with(:trace, false)
+        expect(psettings).to receive(:[]=).with(:evaltrace, false)
```
Comment (DavidS) [inline spec/unit/ace/transport_app_spec.rb]: for completeness sake please add `expect(psettings).not_to receive(:[]=).with(:log_level, :debug)` here and below.

Comment (david22swan) [reply spec/unit/ace/transport_app_spec.rb]: Done

File: spec/unit/ace/transport_app_spec.rb (hunk 1 above)
Comment (david22swan) [reply spec/unit/ace/transport_app_spec.rb]: Done

**Pull request merged**